tritium-processor/src/test/resources/com/palantir/tritium/examples/ProxiedBindsParameter.java.generated

```diff
@@ -0,0 +1,74 @@
+package com.palantir.tritium.examples;
+
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.Method;
+import java.lang.reflect.UndeclaredThrowableException;
+import javax.annotation.processing.Generated;
+
+@Generated("com.palantir.tritium.processor.ProxyAnnotationProcessor")
+public final class ProxiedBindsParameter implements BindsParameter {
+    private static final Method EQUALS;
+
+    private static final Method GET;
+
+    private static final Method TO_STRING;
+
+    static {
+        try {
+            EQUALS = BindsParameter.class.getMethod("equals", Object.class);
```

Comment:
I'm assuming the way this works, is that graal will runs this at build time, and assign static method handles?

Reply:
Not quite, we aren't initializing classes at build time, however compile-time constant reflection calls (in which the target class, method name, and argument classes are all compile-time constants) are automatically replaced by the native-image compiler, so this is safe regardless of how the build is configured. Documentation here: https://www.graalvm.org/reference-manual/native-image/Reflection/#automatic-detection

Reply:
Could we, for V1, detect if we're graaling and build this at compile time?

Reply:
I'm not sure I understand what you're suggesting.

Reply:
Sorry, I indeed was not clear. I was just wondering if you could initialize any of this at build time, so they’re just constants. Not that it matters, but I think normally statics are initialised at runtime?

Reply:
Got it, yes, that could be an improvement.

I think I've vastly overcomplicated this, the same automatic detection can work based on constant arguments to `Proxy.newProxyInstance`, so I just need to generated that based on the `Proxy` annotation. :facepalm: